pkg/build/strategies/onbuild/onbuild.go

```diff
@@ -71,8 +73,34 @@ func (b *OnBuild) SourceTar(config *api.Config) (io.ReadCloser, error) {
 	return b.fs.Open(tarFileName)
 }
 
+func (b *OnBuild) checkAllowedUser(config *api.Config) error {
+	if config.AllowedUIDs == nil || config.AllowedUIDs.Empty() {
+		return nil
+	}
+	user, err := b.docker.GetImageUser(config.BuilderImage)
+	if err != nil {
+		return err
+	}
+	if !userutil.IsUserAllowed(user, &config.AllowedUIDs) {
+		return errors.NewBuilderUserNotAllowedError(config.BuilderImage, false)
+	}
+	cmds, err := b.docker.GetOnBuild(config.BuilderImage)
+	if err != nil {
+		return err
+	}
+	if !userutil.IsOnbuildAllowed(cmds, &config.AllowedUIDs) {
+		return errors.NewBuilderUserNotAllowedError(config.BuilderImage, true)
+	}
+	return nil
+}
+
 // Build executes the ONBUILD kind of build
 func (b *OnBuild) Build(config *api.Config) (*api.Result, error) {
+	// Check whether image user is allowed
+	if err := b.checkAllowedUser(config); err != nil {
```

Comment:
I still need a specific check for onbuild (not only the image user, but also the onbuild instructions)... would it work if I add the check for the image user to strategies.go and then still keep a check in the onbuild strategy?